mne_bids/dig.py

```diff
@@ -188,23 +220,219 @@ def _coordsystem_json(raw, unit, orient, coordsystem_name, fname,
+    """
+    # write electrodes data for iEEG and EEG
+    unit = "m"  # defaults to meters
+
+    params = _parse_bids_filename(electrodes_fname, verbose)
+    subject_id = params['sub']
+    session_id = params['ses']
+    acquisition = params['acq']
+
+    # get coordinate frame from digMontage
+    digpoint = raw.info['dig'][0]
+    if any(digpoint['coord_frame'] != _digpoint['coord_frame']
+           for _digpoint in raw.info['dig']):
+        warn("Not all digpoints have the same coordinate frame. "
+             "Skipping electrodes.tsv writing...")
+    else:
+        # get the accepted mne-python coordinate frames
+        coord_frame_int = int(digpoint['coord_frame'])
+        mne_coord_frame = MNE_FRAME_TO_STR.get(coord_frame_int, None)
+        coord_frame = MNE_TO_BIDS_FRAMES.get(mne_coord_frame, None)
+
+        if verbose:
+            print("Writing electrodes file to... ", electrodes_fname)
+            print("Writing coordsytem file to... ", coordsystem_fname)
+
+        if kind == "ieeg":
+            if coord_frame is not None:
+                # XXX: To improve when mne-python allows coord_frame='unknown'
+                if coord_frame not in BIDS_IEEG_COORDINATE_FRAMES:
+                    coordsystem_fname = make_bids_basename(
+                        subject=subject_id, session=session_id,
+                        acquisition=acquisition, space=coord_frame,
+                        suffix='coordsystem.json', prefix=data_path)
+                    electrodes_fname = make_bids_basename(
+                        subject=subject_id, session=session_id,
+                        acquisition=acquisition, space=coord_frame,
+                        suffix='electrodes.tsv', prefix=data_path)
+                    coord_frame = 'other'
+
+                # Now write the data to the elec coords and the coordsystem
+                _electrodes_tsv(raw, electrodes_fname,
+                                kind, overwrite, verbose)
+                _coordsystem_json(raw, unit, 'n/a',
+                                  coord_frame, coordsystem_fname, kind,
+                                  overwrite, verbose)
+            else:
+                # default coordinate frame to mri if not available
+                warn("Coordinate frame of iEEG coords missing/unknown "
+                     "for {}. Skipping reading "
+                     "in of montage...".format(electrodes_fname))
+        elif kind == 'eeg':
+            # We only write EEG electrodes.tsv and coordsystem.json
+            # if we have LPA, RPA, and NAS available to rescale to a known
+            # coordinate system frame
+            coords = _extract_landmarks(raw.info['dig'])
+            landmarks = set(['RPA', 'NAS', 'LPA']) == set(list(coords.keys()))
+
+            # XXX: to be improved,
+            # mne-python automatically converts unknown coord frame to head
+            if coord_frame_int == FIFF.FIFFV_COORD_HEAD and landmarks:
```

Comment:
Here I would write either
- if coord_frame is HEAD
- or if we have all landmarks (then we rescale to head)

do you agree with this? Does the `XXX` just mean that this approach is currently not possible? what needs to change to make it possible?

Reply:
Doesn't mne-python rescale to head automatically? The XXX is to allow rescaling if there exist landmarks + coordsystem is 'unknown'?

Idk scalp EEG montage that well.

Reply:
I have that issue too in iEEG data, so here was my proposed temporary fix, but I think @agramfort and others chimed in about the difficulties of this since a lot of downstream functionality depends on it. 

https://github.com/mne-tools/mne-python/pull/7709